bundle/regal/ast/ast.rego

```diff
@@ -326,6 +326,24 @@ static_ref(ref) if every t in array.slice(ref.value, 1, count(ref.value)) {
 	t.type != "var"
 }
 
+static_rule_ref(ref) if every t in array.slice(ref, 1, count(ref)) {
+	t.type != "var"
+}
+
+# METADATA
+# description: |
+#   return the name of a rule if, and only if it only has static parts with
+#   no vars. This could be "username", or "user.name", but not "user[name]"
+static_rule_name(rule) := rule.head.ref[0].value if count(rule.head.ref) == 1
+
+static_rule_name(rule) := concat(".", array.concat([rule.head.ref[0].value], [ref.value |
```

Comment:
Would this yield the right thing for
```rego
p["just.like.that"].x := true
```
?

Haven't read to the end, I don't know if it matters here

Reply:
It wouldn't _print_ correctly, but it would work for the purpose of identifying other rules with the same name... which at least at this point will be good enough. But I can add a comment on that.